### ISSUE TITLE
Documentation inaccurate for CrossEntropyLoss (and other loss functions) for multidimensional case

### ISSUE THREAD

**Issue by physicsrob** [pytorch/pytorch#19525]
## 🐛 Bug
This is a simple one:  The documentation for torch.nn.CrossEntropyLoss states that it works for K-dimensional loss where K >= 2.  Reading the documentation I was expecting K=1 to fail, however the code handles K=1 just fine.

It looks like the issue might not be isolated to CrossEntropyLoss.  I'm thinking at minimum the documentation for CrossEntropyLoss, NLLLoss, nll_loss, and cross_entropy all have the bounds wrong for the multi-dimensional case.  The functional losses (nll_loss and cross_entropy) have the bounds right for target but wrong for input.

If there's an agreement that these four cases should be updated, then I'll be happy to submit a PR.

**Comment by soumith** [comment]
yea it looks like a mistake. a PR would be happily accepted, thank you!

**Comment by jbschlosser** [comment]
Fixed in #17300.
